modules/core/src/test/java/org/apache/ignite/internal/processors/cache/GridCachePartitionedClientTxMetricsTest.java

```diff
@@ -0,0 +1,194 @@
+        return super.getConfiguration(igniteInstanceName).setMetricsUpdateFrequency(METRICS_UPDATE_FREQUENCY);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        startGrids(gridCount() - 1);
+
+        startGrid(getConfiguration(getTestIgniteInstanceName(gridCount() - 1)).setClientMode(true));
+
+        awaitPartitionMapExchange();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        super.afterTest();
+
+        for (int i = 0; i < gridCount(); i++)
+            grid(i).cache(DEFAULT_CACHE_NAME).removeAll();
+
+        for (int i = 0; i < gridCount(); i++) {
+            Ignite g = grid(i);
+
+            g.cache(DEFAULT_CACHE_NAME).localMxBean().clear();
+
+            g.transactions().resetMetrics();
+        }
+    }
+
+    /**
+     *
+     */
+    public void testCommitPessimisticReadCommittedTxOnClient() {
+        doPutOnClient(PESSIMISTIC, READ_COMMITTED, true);
+    }
+
+    /**
+     *
+     */
+    public void testCommitPessimisticRepeatableReadTxOnClient() {
+        doPutOnClient(PESSIMISTIC, REPEATABLE_READ, true);
+    }
+
+    /**
+     *
+     */
+    public void testCommitPessimisticSerializableTxOnClient() {
+        doPutOnClient(PESSIMISTIC, SERIALIZABLE, true);
+    }
+
+    /**
+     *
+     */
+    public void testCommitOptimisticReadCommittedTxOnClient() {
+        doPutOnClient(OPTIMISTIC, READ_COMMITTED, true);
+    }
+
+    /**
+     *
+     */
+    public void testCommitOptimisticRepeatableReadTxOnClient() {
+        doPutOnClient(OPTIMISTIC, REPEATABLE_READ, true);
+    }
+
+    /**
+     *
+     */
+    public void testCommitOptimisticSerializableTxOnClient() {
+        doPutOnClient(OPTIMISTIC, SERIALIZABLE, true);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackPessimisticReadCommittedTxOnClient() {
+        doPutOnClient(PESSIMISTIC, READ_COMMITTED, false);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackPessimisticRepeatableReadTxOnClient() {
+        doPutOnClient(PESSIMISTIC, REPEATABLE_READ, false);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackPessimisticSerializableTxOnClient() {
+        doPutOnClient(PESSIMISTIC, SERIALIZABLE, false);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackOptimisticReadCommittedTxOnClient() {
+        doPutOnClient(OPTIMISTIC, READ_COMMITTED, false);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackOptimisticRepeatableReadTxOnClient() {
+        doPutOnClient(OPTIMISTIC, REPEATABLE_READ, false);
+    }
+
+    /**
+     *
+     */
+    public void testRollbackOptimisticSerializableTxOnClient() {
+        doPutOnClient(OPTIMISTIC, SERIALIZABLE, false);
+    }
+
+    /**
+     * @param transactionConcurrency transaction concurrency.
+     * @param transactionIsolation transaction isolation.
```

Comment:
Description should starts with big letter.

Reply:
thx